### PR TITLE
#1013 Fix tests for Windows

### PR DESCRIPTION
**Pull request by pashagolub**
`fcntl` import clause moved from the top and check added for OS name

### CODE REVIEW

File: tests/test_patroni.py [hunk 1]
```diff
@@ -108,11 +109,11 @@ def test_run(self):
         self.p.ha.dcs.watch = Mock(side_effect=SleepException)
         self.p.api.start = Mock()
         self.p.config._dynamic_configuration = {}
-        self.assertRaises(SleepException, self.p.run)
-        with patch('patroni.config.Config.set_dynamic_configuration', Mock(return_value=True)):
-            self.assertRaises(SleepException, self.p.run)
-        with patch('patroni.postgresql.Postgresql.data_directory_empty', Mock(return_value=False)):
-            self.assertRaises(SleepException, self.p.run)
+        # self.assertRaises(SleepException, self.p.run)
```
Comment (CyberDem0n) [inline tests/test_patroni.py]: Why it is commented out?

Comment (pashagolub) [reply tests/test_patroni.py]: Oops! My bad!

Comment (pashagolub) [reply tests/test_patroni.py]: Fixed

File: tests/test_patroni.py [hunk 2]
```diff
@@ -87,8 +87,9 @@ def mock_signal(signo, handler):
         ref = {'passtochild': lambda signo, stack_frame: 0}
 
         def mock_sighup(signo, handler):
-            if signo == signal.SIGHUP:
-                ref['passtochild'] = handler
+            if getattr(signal, 'SIGHUP', None) is not None:
```
Comment (CyberDem0n) [inline tests/test_patroni.py]:
```python
if hasattr(signal, 'SIGHUP') and signo == signal.SIGHUP:
```

File: tests/test_postmaster.py [hunk 3]
```diff
@@ -1,10 +1,21 @@
 import psutil
 import unittest
+import os
```
Comment (CyberDem0n) [inline tests/test_postmaster.py]: Unused import

**Pull request merged**